File: cpp/leetcode/275.cpp
class Solution {
public:
    int hIndex(vector<int>& citations) {
        int l = 0, n = citations.size(), r = len-1,  mid;
        while(l <= r) {
            mid = l + (r - l)/2;
            if(citations[mid] >= (n - mid)) r = mid - 1;
            else l = mid + 1;
        }
        return len - l;
    }
};

File: python/leetcode/275.py
class Solution(object):
    def hIndex(self, citations):
        n = len(citations)
        l, r = 0, n - 1
        while l <= r:
            mid = l + (r - l) / 2
            if n - mid > citations[mid]: l = mid + 1
            else: r = mid - 1
        return n - l